backend/inspirehep/records/api/mixins.py

```diff
@@ -114,242 +112,122 @@ def update_refs_in_citation_table(self, save_every=100):
-
-        Returns:
-            str: Key of downloaded file, or `None` if file was not found
-        """
-        urls = [_url for _url in [url, original_url] if _url]
-        key = None
-        for _url in urls:
-            try:
-                if _url.startswith(FILES_API_PREFIX):
-                    LOGGER.debug("file is local", url=url)
-                    key = self.download_file_from_local_storage(_url)
-            except FileNotFoundError:
-                LOGGER.warning("Cannot copy from local storage", url=url)
-        if not key:
-            for _url in urls:
-                try:
-                    if _url.startswith("http"):
-                        LOGGER.debug("file is web based.", url=url)
-                        key = self.download_file_from_url(_url)
-                except ResourceNotFoundError:
-                    LOGGER.warning("Cannot download file", url=url)
-        return key
-
-    def download_file_from_local_storage(self, url, **kwargs):
-        """Opens local file with ObjectVersion API.
-        Callculates it's hash and returns hash of the file.
-
-        Args:
-            url (str): Local url which starts with /api/files/
-
-        Returns:
-            str: key(sha-1) of downloaded file
+        if not file_object:
+            return None
 
-        Examples:
-            >>> url = '/api/files/261926f6-4923-458e-adb0/207611e7bf8a83f0739bb2e'
-            >>> self.download_file_from_local_storage(url)
-                '207611e7bf8a83f0739bb2e'
-        """
-        url_splited = self.split_url(url)
-        try:
-            file_ = self.find_local_file(
-                key=url_splited["file"], bucket_id=url_splited["bucket"]
-            )
-        except ValueError:
-            LOGGER.exception("Cannot download file from local storage", url=url_splited)
-            file_ = None
+        key_hashed = key if self.is_hash(key) else None
 
-        if not file_:
-            raise FileNotFoundError(f"{url} is not a valid file in local storage!")
-        file_hash = None
-        if self.is_hash(url_splited["file"]):
-            file_hash = url_splited["file"]
-        return self.copy_local_file(file_, file_hash)
+        if not self.verify_hash_of_files(file_object, key_hashed):
+            key_hashed = self.hash_data(file_instance=file_object)
 
-    def download_file_from_url(self, url):
-        """Downloads file and calculates hash for it
+        if key_hashed not in self.files.keys:
+            file_object.copy(bucket=self.files.bucket.id, key=key_hashed)
 
-        If everything is ok then adds it to files in current record.
-        If file with same hash already found in db, tries to use this one
-        instead of creating duplicate (uses `ObjectVersion.copy()` method)
+        self.files.flush()
+        return key_hashed
 
-        Args:
-            url (str): Local or remote url/filepath
+    def add_file_from_url(self, url):
```

Comment:
If I understand correctly but it's handled and there is a test for this https://github.com/inspirehep/inspirehep/blob/c731ebab84a2af86173d2d510ed6c7e51b6652d9/backend/tests/integration/records/api/test_api_mixins.py#L64

Reply:
No, this test checks local file only so actually we know the key as it's in the local url already.
I'm writing about situation when you have only url (like during harvest) and actually file is already in other record. Now we will add it again in another bucket, and requirement was to just to use copy method which actually won't copy data but it will increase reference count for it.